crates/router/src/core/customers.rs

```diff
@@ -16,62 +19,125 @@ use crate::{
     services,
     types::{
         api::customers,
+        domain::{
+            self,
+            types::{self, AsyncLift, TypeEncryption},
+        },
         storage::{self, enums},
     },
+    utils::generate_id,
 };
 
 pub const REDACTED: &str = "Redacted";
 
 #[instrument(skip(db))]
 pub async fn create_customer(
     db: &dyn StorageInterface,
-    merchant_account: storage::MerchantAccount,
+    merchant_account: domain::MerchantAccount,
     mut customer_data: customers::CustomerRequest,
 ) -> RouterResponse<customers::CustomerResponse> {
     let customer_id = &customer_data.customer_id;
     let merchant_id = &merchant_account.merchant_id;
     customer_data.merchant_id = merchant_id.to_owned();
 
+    let key = types::get_merchant_enc_key(db, merchant_id.clone())
+        .await
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Failed while getting encryption key")?;
+
     if let Some(addr) = &customer_data.address {
         let customer_address: api_models::payments::AddressDetails = addr
             .peek()
             .clone()
             .parse_value("AddressDetails")
-            .change_context(errors::ApiErrorResponse::InvalidDataValue {
-                field_name: "address",
-            })?;
-        db.insert_address(storage::AddressNew {
-            city: customer_address.city,
-            country: customer_address.country,
-            line1: customer_address.line1,
-            line2: customer_address.line2,
-            line3: customer_address.line3,
-            zip: customer_address.zip,
-            state: customer_address.state,
-            first_name: customer_address.first_name,
-            last_name: customer_address.last_name,
-            phone_number: customer_data.phone.clone(),
-            country_code: customer_data.phone_country_code.clone(),
-            customer_id: customer_id.to_string(),
-            merchant_id: merchant_id.to_string(),
-            ..Default::default()
-        })
+            .change_context(errors::ApiErrorResponse::AddressNotFound)?;
```

Comment:
This change raises an incorrect error message that the address was not found, when in reality, the deseriliazation of the address failed. Could you please fix this, and ensure that "resource not found" errors are not being raised incorrectly elsewhere?